### PR TITLE
expression: Fix arg validation for tidb_mvcc_info()

### LINKED ISSUE #58309: connection running loop panic: select tidb_mvcc_info()
### 1. Minimal reproduce step (Required)

```
select tidb_mvcc_info(); do 1;
```

### 2. What did you expect to see? (Required)

```
ERROR 1105 (HY000): unexpected length of args and argTps
Query OK, 0 rows affected (0.01 sec)

```

### 3. What did you see instead (Required)

```
ERROR 1105 (HY000): unexpected length of args and argTps
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    2097160
Current database: test

Query OK, 0 rows affected (0.01 sec)

```

### 4. What is your TiDB version? (Required)

```
Release Version: v8.5.0-alpha-347-gcc83417e59
Edition: Community
Git Commit Hash: cc83417e5937bd030247c1bf0cb5dc2b2367555b
Git Branch: master
UTC Build Time: 2024-12-16 17:12:55
GoVersion: go1.23.4
Race Enabled: false
Check Table Before Drop: false
Store: unistore
```

### Stack

```
[2024/12/16 18:17:28.549 +01:00] [ERROR] [conn.go:1041] ["connection running loop panic"] [conn=2097158] [session_alias=] [lastSQL="select tidb_mvcc_info()"] [err="unexpected length of args and argTps"] [stack="github.com/pingcap/tidb/pkg/server.(*clientConn).Run.func1\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1044\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:785\ngithub.com/pingcap/tidb/pkg/executor.(*Compiler).Compile.func1\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/executor/compiler.go:58\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:785\ngithub.com/pingcap/tidb/pkg/expression.newBaseBuiltinFuncWithTp\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/expression/builtin.go:187\ngithub.com/pingcap/tidb/pkg/expression.(*tidbMVCCInfoFunctionClass).getFunction\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/expression/builtin_info.go:929\ngithub.com/pingcap/tidb/pkg/expression.newFunctionImpl\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/expression/scalar_function.go:265\ngithub.com/pingcap/tidb/pkg/expression.NewFunction\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/expression/scalar_function.go:314\ngithub.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).newFunctionWithInit\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:1640\ngithub.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).newFunction\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:1650\ngithub.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).funcCallToExpression\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:2405\ngithub.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).Leave\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:1474\ngithub.com/pingcap/tidb/pkg/parser/ast.(*FuncCallExpr).Accept\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/parser/ast/functions.go:598\ngithub.com/pingcap/tidb/pkg/planner/core.rewriteExprNode\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:300\ngithub.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).rewriteWithPreprocess\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:239\ngithub.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).buildProjection\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/logical_plan_builder.go:1337\ngithub.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).buildSelect\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/logical_plan_builder.go:3888\ngithub.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).Build\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/planbuilder.go:543\ngithub.com/pingcap/tidb/pkg/planner.buildLogicalPlan\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/optimize.go:575\ngithub.com/pingcap/tidb/pkg/planner.optimize\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/optimize.go:492\ngithub.com/pingcap/tidb/pkg/planner.Optimize\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/planner/optimize.go:360\ngithub.com/pingcap/tidb/pkg/executor.(*Compiler).Compile\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/executor/compiler.go:101\ngithub.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/session/session.go:2099\ngithub.com/pingcap/tidb/pkg/server.(*TiDBContext).ExecuteStmt\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/driver_tidb.go:291\ngithub.com/pingcap/tidb/pkg/server.(*clientConn).handleStmt\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:2029\ngithub.com/pingcap/tidb/pkg/server.(*clientConn).handleQuery\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1782\ngithub.com/pingcap/tidb/pkg/server.(*clientConn).dispatch\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1378\ngithub.com/pingcap/tidb/pkg/server.(*clientConn).Run\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1147\ngithub.com/pingcap/tidb/pkg/server.(*Server).onConn\n\t/home/dvaneeden/dev/pingcap/tidb/pkg/server/server.go:741"]
```

```
github.com/pingcap/tidb/pkg/server.(*clientConn).Run.func1
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1044
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:785
github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile.func1
	/home/dvaneeden/dev/pingcap/tidb/pkg/executor/compiler.go:58
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:785
github.com/pingcap/tidb/pkg/expression.newBaseBuiltinFuncWithTp
	/home/dvaneeden/dev/pingcap/tidb/pkg/expression/builtin.go:187
github.com/pingcap/tidb/pkg/expression.(*tidbMVCCInfoFunctionClass).getFunction
	/home/dvaneeden/dev/pingcap/tidb/pkg/expression/builtin_info.go:929
github.com/pingcap/tidb/pkg/expression.newFunctionImpl
	/home/dvaneeden/dev/pingcap/tidb/pkg/expression/scalar_function.go:265
github.com/pingcap/tidb/pkg/expression.NewFunction
	/home/dvaneeden/dev/pingcap/tidb/pkg/expression/scalar_function.go:314
github.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).newFunctionWithInit
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:1640
github.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).newFunction
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:1650
github.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).funcCallToExpression
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:2405
github.com/pingcap/tidb/pkg/planner/core.(*expressionRewriter).Leave
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:1474
github.com/pingcap/tidb/pkg/parser/ast.(*FuncCallExpr).Accept
	/home/dvaneeden/dev/pingcap/tidb/pkg/parser/ast/functions.go:598
github.com/pingcap/tidb/pkg/planner/core.rewriteExprNode
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:300
github.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).rewriteWithPreprocess
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/expression_rewriter.go:239
github.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).buildProjection
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/logical_plan_builder.go:1337
github.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).buildSelect
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/logical_plan_builder.go:3888
github.com/pingcap/tidb/pkg/planner/core.(*PlanBuilder).Build
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/core/planbuilder.go:543
github.com/pingcap/tidb/pkg/planner.buildLogicalPlan
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/optimize.go:575
github.com/pingcap/tidb/pkg/planner.optimize
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/optimize.go:492
github.com/pingcap/tidb/pkg/planner.Optimize
	/home/dvaneeden/dev/pingcap/tidb/pkg/planner/optimize.go:360
github.com/pingcap/tidb/pkg/executor.(*Compiler).Compile
	/home/dvaneeden/dev/pingcap/tidb/pkg/executor/compiler.go:101
github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt
	/home/dvaneeden/dev/pingcap/tidb/pkg/session/session.go:2099
github.com/pingcap/tidb/pkg/server.(*TiDBContext).ExecuteStmt
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/driver_tidb.go:291
github.com/pingcap/tidb/pkg/server.(*clientConn).handleStmt
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:2029
github.com/pingcap/tidb/pkg/server.(*clientConn).handleQuery
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1782
github.com/pingcap/tidb/pkg/server.(*clientConn).dispatch
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1378
github.com/pingcap/tidb/pkg/server.(*clientConn).Run
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/conn.go:1147
github.com/pingcap/tidb/pkg/server.(*Server).onConn
	/home/dvaneeden/dev/pingcap/tidb/pkg/server/server.go:741
```

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #58309 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
A situation where the connection was dropped when TIDB_MVCC_INFO() was called without arguments was fixed.
```

### Related

- https://github.com/pingcap/docs/pull/19715

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

File: pkg/parser/parser_test.go [hunk 1]
```diff
@@ -1693,7 +1693,6 @@ func TestBuiltin(t *testing.T) {
 		{`SELECT tidb_decode_key('abc');`, true, "SELECT TIDB_DECODE_KEY(_UTF8MB4'abc')"},
 		{`SELECT tidb_decode_base64_key('abc');`, true, "SELECT TIDB_DECODE_BASE64_KEY(_UTF8MB4'abc')"},
 		{`SELECT tidb_decode_sql_digests('[]');`, true, "SELECT TIDB_DECODE_SQL_DIGESTS(_UTF8MB4'[]')"},
-		{`SELECT get_mvcc_info('hex', '0xabc');`, true, "SELECT GET_MVCC_INFO(_UTF8MB4'hex', _UTF8MB4'0xabc')"},
```
Comment (D3Hunter) [inline pkg/parser/parser_test.go:1696]: why remove it

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Comment by dveeden** [comment]
> Are there any way to have a test for this change?

Done in 48504c5326e7cf3ddc905191901079f26b1adf3b

**Review by D3Hunter — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**